src/EditorFeatures/CSharpTest/SpellCheck/SpellCheckSpanTests.cs

```diff
@@ -0,0 +1,517 @@
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier16()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    private int {|Identifier:X|} => 0;
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier17()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    private event Action {|Identifier:X|};
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier18()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    private event Action {|Identifier:X|}, {|Identifier:Y|};
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier19()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    private event Action {|Identifier:X|} { add { } remove { } }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier20()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}()
+    {
+        int {|Identifier:E|};
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier21()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}()
+    {
+        int {|Identifier:E|}, {|Identifier:F|};
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier22()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}()
+    {
+{|Identifier:E|}:
+        return;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier23()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}(int {|Identifier:E|})
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier24()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}(int {|Identifier:E|})
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier25()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}(int {|Identifier:E|}, int {|Identifier:F|})
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier26()
+        {
+            await TestAsync(@"
+static class {|Identifier:C|}
+{
+    static void {|Identifier:D|}(this int {|Identifier:E|})
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier27()
+        {
+            await TestAsync(@"
+namespace {|Identifier:C|}
+{
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier28()
+        {
+            await TestAsync(@"
+namespace {|Identifier:C|}.{|Identifier:D|}
+{
+}");
+        }
+
+        [Fact]
+        public async Task TestIdentifier29()
+        {
+            await TestAsync(@"
+class {|Identifier:C|}
+{
+    void {|Identifier:D|}()
+    {
+        for (int {|Identifier:E|} = 0; E < 10; E++)
```

Comment:
Why aren't the other `E`s on this line identifiers? Do we just not classify them as as `LocalName`? Same for `Goo` below? I thought we classified method declarations, and calls as the same thing?

I was expecting to see something in the code that only reported each identifier once or something, but unless I missed it, that doesn't seem to be happening

Reply:
so the idea of 'identifier' is taht it's really 'the declaration location of a symbol name'.  so in the cases you brought up, those are references, and this is showing correctly that we do not spell check those.  We only spell check a symbol at the point of declaration.

Reply:
Sure, that completely makes sense, I just don't see in the code where that is handled

Reply:
it's how hte syntactic classifier works (which we call into).  it can't classify a random reference (since it has no idea waht it is).  It can only classify decls as that's hte only location you know what that identifier token means.

Reply:
Thank you, thats what i was missing - its the syntactic classifier, not semantic.